spring-cloud-gcp-data-firestore/src/test/java/org/springframework/cloud/gcp/data/firestore/FirestoreTemplateTests.java

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright 2019-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.data.firestore;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import com.google.firestore.v1.CreateDocumentRequest;
+import com.google.firestore.v1.DeleteDocumentRequest;
+import com.google.firestore.v1.Document;
+import com.google.firestore.v1.FirestoreGrpc.FirestoreStub;
+import com.google.firestore.v1.RunQueryRequest;
+import com.google.firestore.v1.RunQueryResponse;
+import com.google.firestore.v1.StructuredQuery;
+import com.google.firestore.v1.Value;
+import com.google.protobuf.Empty;
+import io.grpc.stub.StreamObserver;
+import org.junit.Before;
+import org.junit.Test;
+
+import org.springframework.data.annotation.Id;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+/**
+ * @author Dmitry Solomakha
+ * @since 1.2
+ */
+public class FirestoreTemplateTests {
+	private FirestoreTemplate firestoreTemplate;
+
+	private final FirestoreStub firestoreStub = mock(FirestoreStub.class);
+
+	private final String parent = "projects/my-project/databases/(default)/documents";
+
+	@Before
+	public void setup() {
+		this.firestoreTemplate = new FirestoreTemplate(this.firestoreStub, this.parent);
+	}
+
+	@Test
+	public void saveTest() {
+		TestEntity testEntity1 = new TestEntity("test_entity_1", 123L);
+
+		doAnswer(invocation -> {
+			StreamObserver<Document> streamObserver = invocation.getArgument(1);
+			streamObserver.onNext(Document.newBuilder().build());
+			streamObserver.onCompleted();
+			return null;
+		}).when(this.firestoreStub).createDocument(any(), any());
+
+		this.firestoreTemplate.save(testEntity1).block();
```

Comment:
`save()` returns a value -- can that be validated?

Reply:
done